File: WidgetsDemo/Spinner/app/src/main/java/com/example/spinner/MainActivity.java
package com.example.spinner;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.AdapterView.OnItemSelectedListener;
import android.widget.ArrayAdapter;
import android.widget.Spinner;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;
import java.util.Collections;

import static com.example.spinner.R.layout.activity_main;
import static com.example.spinner.R.layout.support_simple_spinner_dropdown_item;
import static java.lang.String.*;
import static java.util.Arrays.asList;
import static java.util.Collections.*;

public class MainActivity extends AppCompatActivity {
	Spinner spinner;
	TextView textView;
	ArrayList<String> stringArrayList;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(activity_main);
		spinner = findViewById(R.id.spinner);
		textView = findViewById(R.id.textView);
		stringArrayList = new ArrayList<>();
		addAll(stringArrayList, "", "Tim", "Jim", "Kim", "Pim");
		spinner.setAdapter(new ArrayAdapter<>(this, support_simple_spinner_dropdown_item, stringArrayList));
		spinner.setOnItemSelectedListener(new OnItemSelectedListener() {
			@Override
			public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
				textView.setText(format("Selection: %s", stringArrayList.get(position)));
			}

			@Override
			public void onNothingSelected(AdapterView<?> parent) {

			}
		});
	}
}

File: JSONDemo/app/src/main/java/com/example/jsondemo/MainActivity.kt
package com.example.jsondemo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.jsondemo.R.layout.activity_main
import org.json.JSONException
import org.json.JSONObject
import java.util.*

class MainActivity : AppCompatActivity() {
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(activity_main)
		/*val schoolInfo = JSONObject()
		try {
			schoolInfo.put("name", "Chirag Baviskar").put("grade", 11).put("id", 30000054)
		} catch (e: JSONException) {
			e.printStackTrace()
		}
		val compSciClass = JSONObject()
		try {
			compSciClass.put("compscigrade", "A").put("compscipercent", 95)
			schoolInfo.put("comp sci class", compSciClass)
		} catch (e: JSONException) {
			e.printStackTrace()
		}
		val bioClass = JSONObject()
		try {
			bioClass.put("biograde", "A-").put("biopercent", 90)
			schoolInfo.put("bio class", compSciClass)
		} catch (e: JSONException) {
			e.printStackTrace()
		}
		d("TAG", schoolInfo.toString())
		try {
			d("RETRIEVE", String.format("%s (%s)", schoolInfo.getJSONObject("bio class").getString("biograde"), schoolInfo.getJSONObject("bio class").getString("biopercent")))
		} catch (e: JSONException) {
			e.printStackTrace()
		}*/
		val music = JSONObject();
		val stringList = ArrayList(listOf("Mark Ronson Featuring Bruno Mars", "Uptown Funk", "LMFAO Featuring Lauren Bennett & GoonRock", "Party Rock Anthem", "Ed Sheeran", "Shape Of You", "The Chainsmokers Featuring Halsey", "Closer", "Maroon 5 Featuring Cardi B", "Girls Like You"))
		var i = 0
		while (i < stringList.size) {
			try {
				music.put("value $i", JSONObject().put("artist $i", stringList[i]).put("song $i", stringList[i + 1]))
			} catch (e: JSONException) {
				e.printStackTrace()
			}
			i += 2
		}
	}
}